apollo-client/src/main/java/com/ctrip/framework/apollo/spi/DefaultConfigFactory.java

```diff
@@ -43,19 +44,21 @@
  */
 public class DefaultConfigFactory implements ConfigFactory {
   private static final Logger logger = LoggerFactory.getLogger(DefaultConfigFactory.class);
-  private ConfigUtil m_configUtil;
+  private final ConfigUtil m_configUtil;
+  private final RepositoryConfigFactory m_repositoryConfigFactory;
 
   public DefaultConfigFactory() {
     m_configUtil = ApolloInjector.getInstance(ConfigUtil.class);
+    m_repositoryConfigFactory = ApolloInjector.getInstance(RepositoryConfigFactory.class);
   }
 
   @Override
   public Config create(String namespace) {
     ConfigFileFormat format = determineFileFormat(namespace);
     if (ConfigFileFormat.isPropertiesCompatible(format)) {
-      return new DefaultConfig(namespace, createPropertiesCompatibleFileConfigRepository(namespace, format));
+      return this.m_repositoryConfigFactory.createConfig(namespace, createPropertiesCompatibleFileConfigRepository(namespace, format));
```

Comment:
I'm not sure whether it's necessary to add another abstraction of RepositoryConfigFactory. Is it simpler to write a custom config factory in `apollo-client-config-data` module which extends DefaultConfigFactory and only overrides the `create` method, then we could use the ApolloConfigDataInjectorCustomizer to customize the injector?

Reply:
resolved